base/operators.jl

```diff
@@ -1347,13 +1347,11 @@ end
 
 # Specialized variant of in for Tuple, which can generate typed comparisons for each element
 # of the tuple, skipping values that are statically known to be != at compile time.
-function in(x, itr::Tuple)
-    @_terminates_globally
-    _in_tuple(x, itr, false)
-end
+in(x, itr::Tuple) = _in_tuple(x, itr, false)
 # This recursive function will be unrolled at compiletime, and will not generate separate
 # llvm-compiled specializations for each step of the recursion.
-@inline function _in_tuple(x, @nospecialize(itr::Tuple), anymissing::Bool)
+function _in_tuple(x, @nospecialize(itr::Tuple), anymissing::Bool)
+    @inline
```

Comment:
Can you help me understand why you moved the inline annotation? Does it have any difference, or you just prefer this stylistically? Thanks!

Reply:
1. An `@inline` applied to a method carries less weight than an `@inline` applied to code block within the method body.

2. The semantics are different, when used within the method body it suggests to inline callees.

Reply:
BTW can you help review my linked PR, given that you're freshly acquainted with the code?

Reply:
They have the same effect, but `@inline method def` is defined in base/expr.jl and is not available at this point of loading base/operators.jl when building system image. the version which is used without any arguments on the other hand is defined at the very early stage of sysimg build, so we can use it even before loading base/expr.jl.